lib/io/hardware_gpio.hpp

```diff
@@ -2,31 +2,63 @@
 
 #include "gpio.hpp"
 
+#include <unordered_map>
+
 #include <core/types.hpp>
 
 namespace hyped::io {
 
+// GPIO hardware specified addresses and sizes for read, clear, set, size, etc.
+static constexpr std::uint32_t pin_size  = 0x1000;
+static constexpr std::uint32_t pin_read  = 0x138;
+static constexpr std::uint32_t pin_clear = 0x190;
+static constexpr std::uint32_t pin_set   = 0x194;
```

Comment:
these become `private static constexpr`s following the style guide discussion

unless they need to be accessed outside of methods